tests/components/konnected/test_config_flow.py

```diff
@@ -0,0 +1,1052 @@
+            {"activation": "high", "zone": "alarm1"},
+        ],
+    }
+
+
+async def test_option_flow_import(hass, mock_panel):
+    """Test config flow options imported from configuration.yaml."""
+    device_options = config_flow.OPTIONS_SCHEMA(
+        {
+            "io": {
+                "1": "Binary Sensor",
+                "2": "Digital Sensor",
+                "3": "Switchable Output",
+            },
+            "binary_sensors": [
+                {"zone": "1", "type": "window", "name": "winder", "inverse": True},
+            ],
+            "sensors": [{"zone": "2", "type": "ds18b20", "name": "temper"}],
+            "switches": [
+                {
+                    "zone": "3",
+                    "name": "switcher",
+                    "activation": "low",
+                    "momentary": 50,
+                    "pause": 100,
+                    "repeat": 4,
+                },
+            ],
+        }
+    )
+
+    device_config = config_flow.CONFIG_ENTRY_SCHEMA(
+        {
+            "host": "1.2.3.4",
+            "port": 1234,
+            "id": "112233445566",
+            "model": "Konnected Pro",
+            "access_token": "11223344556677889900",
+            "default_options": device_options,
+        }
+    )
+
+    entry = MockConfigEntry(
+        domain="konnected", data=device_config, unique_id="112233445566"
+    )
+    entry.add_to_hass(hass)
+
+    result = await hass.config_entries.options.async_init(
+        entry.entry_id, context={"source": "test"}
+    )
+    assert result["type"] == "form"
+    assert result["step_id"] == "options_io"
+
+    # confirm the defaults are set based on current config - we"ll spot check this throughout
+    schema = result["data_schema"]({})
+    assert schema["1"] == "Binary Sensor"
+    assert schema["2"] == "Digital Sensor"
+    assert schema["3"] == "Switchable Output"
+
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={
+            "1": "Binary Sensor",
+            "2": "Digital Sensor",
+            "3": "Switchable Output",
+        },
+    )
+    assert result["type"] == "form"
+    assert result["step_id"] == "options_io_ext"
+    schema = result["data_schema"]({})
+    assert schema["8"] == "Disabled"
+
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"], user_input={},
+    )
+    assert result["type"] == "form"
+    assert result["step_id"] == "options_binary"
+
+    # zone 1
+    schema = result["data_schema"]({})
+    assert schema["type"] == "window"
+    assert schema["name"] == "winder"
+    assert schema["inverse"] is True
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"], user_input={"type": "door"}
+    )
+    assert result["type"] == "form"
+    assert result["step_id"] == "options_digital"
+
+    # zone 2
+    schema = result["data_schema"]({})
+    assert schema["type"] == "ds18b20"
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=4986898)